### DISCUSSION TITLE
What is the reason of having to sync actor state with spawning machine?

### DISCUSSION THREAD

**Discussion by geekox86** [statelyai/xstate#2346] [Q&A] [answered]
Hi again.

I am just wondering why is this the case?
Is it related to a limitation in serializing the machine context?
Or is it a deeper principle related to state encapsulation?

I find it a little counter-intuitive as the actor state is directly accessible using useActor from outside the machine, so why not inside the machine?

Any specific changes to this behavior in v5?

**Comment by davidkpiano** [comment, marked as answer]
The main question was confusing so I'll answer this one:

> I find it a little counter-intuitive as the actor state is directly accessible using useActor from outside the machine, so why not inside the machine?

Don't think of it as accessing actor state; you're accessing the _last emitted value_ from the actor, which can represent its state (as it usually does). The distinction is important because actors can live anywhere, so it's always possible that the emitted value might be stale (eventually consistent).

Anyway, in the latest version of v4, you can read this via `someActorRef.getSnapshot()`.

**Reply by geekox86** [reply]
Why not keep a reference to the actor and access its state as usual from inside the parent just like one would with any other service?

**Reply by davidkpiano** [reply]
Can you give an example of accessing state "as usual from inside the parent"? What do you mean by that?

**Reply by geekox86** [reply]
Hi again. Sorry, for not being clear.

If I want to access the actor state from outside the machine that originally created the actor, I would do something like this:
`const { state } = useActor(service.state.context.someActor);`
This seems to me access as usual.

My question, why can't access the same actor from inside the machine, for example, like this:
`(context) => context.someActor.state`
I mean by reference.

**Reply by davidkpiano** [reply]
You can using `.getSnapshot()`:

```js
(context) => context.someActor.getSnapshot() // this is the state
```

**Reply by geekox86** [reply]
Out of curiosity, does `useActor` subscribe to changes form `service.state.context.someActor` and update the returned `{ state }` using `getSnapshot()`?

**Reply by davidkpiano** [reply]
Yes, it does.

**Reply by geekox86** [reply]
Thank you sir :)

**Reply by geekox86** [reply]
@davidkpiano sorry for opening this subject again but I have a question :)
How is spawning an actor different than simply assigning an interpreted machine to the context?
In the later case I will also have access to the state without the need to call getSnapshot.

I can only see two limitations here: serialization (which also exists with actors) and lack of sendParent/respond in interpreted machine?
